runtime/src/main/java/org/wildfly/swarm/config/runtime/invocation/SimpleTypeAdapter.java

```diff
@@ -113,6 +113,23 @@ else if(String.class.equals(propertyType))
             else
                 value = "";
         }
+        else if(Enum.class.isAssignableFrom(propertyType))
+        {
+            if(dmrPayload.isDefined()) {
+                final String allowedValue = dmrPayload.asString();
+                // The toString() value should be the same as the DMR string
+                for (Object e : propertyType.getEnumConstants()) {
+                    if (allowedValue.equals(e.toString())) {
+                        value = e;
+                        break;
+                    }
+                }
+                if (value == null) {
+                    throw new RuntimeException(String.format("Could not determine correct enum value for type %s with value %s", propertyType, dmrPayload.asString()));
+                }
```

Comment:
The value is scoped by the 'allowed-values' meta data, so falling back to a simple string doesn't make much sense. I think your approach to fail fast is in this case desireable.